pyfar/signals/deterministic.py

```diff
@@ -255,6 +391,474 @@ def exponential_sweep_time(n_samples, frequency_range, n_fade_out=90,
+        maximum of the sweep time signal. The default ``0`` does not apply
+        a fade-in.
+    fade_out : int
+        Duration of a squared cosine fade-out in samples. The fade ends at the
+        last sample of the sweep that is closer than 60 dB to the absolute
+        maximum of the sweep time signal. The default ``0`` does not apply
+        a fade-out.
+    butterworth_order : int, None
+        The order of the Butterworth filters that are applied to limit the
+        frequency range by a high-pass if ``frequency_range[0] > 0`` and/or by
+        a low-pass if ``frequency_range[1] < sampling_rate / 2``. The default
+        is ``8``.
+    sampling_rate : int
+        The sampling rate in Hz. The default is ``44100``.
+
+    Returns
+    -------
+    sweep : Signal
+        The sweep signal. The Signal is in the time domain, has a maximum
+        absolute amplitude of 1 and the ``rms`` FFT normalization
+        (see :py:func:`~pyfar.dsp.fft.normalization`).
+    group_delay_sweep : FrequencyData
+        The group delay of the sweep in seconds as a single sided spectrum.
+
+    Notes
+    -----
+    The envelope of the sweep time signal should be constant, apart from
+    slight overshoots at the beginning and end. If this is not the case, try
+    to increase `n_samples`, `start_margin`, `stop_margin`, `fade_in` or
+    `fade_out`.
+
+    References
+    ----------
+    .. [#] S. Müller, P. Massarani. 'Transfer Function Measurement with Sweeps.
+           Directors Cut Including Previously Unreleased Material and some
+           Corrections. J. Audio Eng. Soc. 2001, 49 (6), 443-471.
+
+    Examples
+    --------
+
+    Exponential sweep between 50 and 22050 Hz
+
+    .. plot::
+
+        >>> import pyfar as pf
+        >>> sweep, _ = pf.signals.exponential_sweep_freq(
+        ...     2**16, [50, 22050], 5000, 1000)
+        >>> pf.plot.time_freq(sweep)
+    """
+
+    signal, group_delay = _frequency_domain_sweep(
+        n_samples=n_samples,
+        sweep_type='exponential',
+        frequency_range=frequency_range,
+        butterworth_order=butterworth_order,
+        start_margin=start_margin,
+        stop_margin=stop_margin,
+        fade_in=fade_in,
+        fade_out=fade_out,
+        sampling_rate=sampling_rate)
+
+    return signal, group_delay
+
+
+def magnitude_spectrum_weighted_sweep(
+        n_samples, magnitude_spectrum, start_margin, stop_margin,
+        fade_in=None, fade_out=None, sampling_rate=44100):
+    """
+    Generate single channel sine sweep with arbitrary magnitude spectrum.
+
+    Sweep sweep synthesis according to [#]_
+
+    .. note::
+        The linear sweep can also be generated in the time domain
+        (:py:func:`~linear_sweep_time`). Frequency domain synthesis exhibits
+        smooth magnitude spectra in trade of a slightly irregular temporal
+        envelope. Time domain synthesis exhibits a constant temporal envelope
+        in trade of slight ripples in the magnitude response.
```

Comment:
Ups. I think this was a copy paste thing. I adapted it according to your comment. Considering that the instantanoud frequency of a sine signal designed in the time domain is given by the derivate of its argument, it might be possible to have a time domain equivalent - but I'm not aware of any work on this.